### PR TITLE
feat: minimal PWA manifest

### PR DESCRIPTION
**Pull request by lidel**
This PR is the first stab at making it possible to install and run in app mode.

### TODO

- [x] manifest and icon
- [x] land other things from the IPFS Camp milestone
- [x] rebase and tweak any papercuts 
- [ ] add protocol handlers and `?uri=` router (details in #21)

Aims to work towards #21

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "short_name": "InBrowser IPFS",
+  "name": "InBrowser IPFS Gateway",
```
Comment (lidel) [inline public/manifest.json:3]: :thought_balloon: ideas for better name welcome. We need something short, perhaps just go with `inbrowser.link` ?

**Review by SgtPooki — COMMENTED** [review]
Code looks fine to me but i havent looked at a pwa manifest in a long time. Ill need to deploy local and see if i can get it to work

File: public/manifest.json (hunk 1 above)
Comment (SgtPooki) [reply public/manifest.json:3]: i think thats about as short as we can go and have it still be semantic

**Comment by lidel** [comment]
@SgtPooki it is super tricky to test on mobile in LAN due to lack of https, but I was able to do it in desktop Google Chrome with `goo.localhost:3000`, it shows small icon on the right side of address bar that allows installing the PWA. 

Something that is missing is [`protocol_handlers`](https://developer.mozilla.org/en-US/docs/Web/Manifest/protocol_handlers), but we need and `?uri=%s` router ([specs](https://specs.ipfs.tech/http-gateways/subdomain-gateway/#uri-router)) for that.

I am thinking I could add router by converting `?uri=%s` into `/ipxs/id/..` path before passing it to `verifiedFetch` [here](https://github.com/ipfs/service-worker-gateway/blob/774e388bd2c4696988121c42ce511c59837348f0/src/sw.ts#L492), but lmk if there is a better way?
It could also be implemented as redirect response upstream in verified fetch, but not sure how useful it is beyond this project.

**Comment by SgtPooki** [comment]
> @SgtPooki it is super tricky to test on mobile in LAN due to lack of https, but I was able to do it in desktop Google Chrome with goo.localhost:3000, it shows small icon on the right side of address bar that allows installing the PWA.

Mac allows you to do simulation of iphone from desktop, i will give this a quick try before i move over to something else

**Comment by SgtPooki** [comment]
> > @SgtPooki it is super tricky to test on mobile in LAN due to lack of https, but I was able to do it in desktop Google Chrome with goo.localhost:3000, it shows small icon on the right side of address bar that allows installing the PWA.
> 
> Mac allows you to do simulation of iphone from desktop, i will give this a quick try before i move over to something else

image

image

image

image

image

**Comment by SgtPooki** [comment]
there are some UX issues and issues with content for specs.ipfs.tech. The simulator history back and forward work, but it's hard to trigger.. in general it seems to work okay!

**Comment by lidel** [comment]
Thanks! Let's ship this, and add protocol handlers at a later time.

**Pull request merged**